main/lsp/requests/rename.cc

```diff
@@ -29,48 +30,187 @@ bool isValidRenameLocation(const core::SymbolRef &symbol, const core::GlobalStat
                 fmt::format("Renaming constants defined in {} files is not supported; symbol {} is defined at {}",
                             filetype, symbol.data(gs)->name.show(gs), loc.filePosToString(gs));
             response->error = make_unique<ResponseError>((int)LSPErrorCodes::InvalidRequest, error);
-
             return false;
         }
     }
     return true;
 }
-} // namespace
 
-variant<JSONNullObject, unique_ptr<WorkspaceEdit>>
-RenameTask::getRenameEdits(LSPTypecheckerDelegate &typechecker, core::SymbolRef symbol, std::string_view newName) {
-    const core::GlobalState &gs = typechecker.state();
-    vector<unique_ptr<Location>> references = getReferencesToSymbol(typechecker, symbol);
-    auto originalName = symbol.data(gs)->name.toString(gs);
-    auto we = make_unique<WorkspaceEdit>();
+// Replaces originalName with newName in the last dotted component if the string is of the form x.y.z; otherwise does a
+// simple ReplaceAll.
+string replaceLastDotted(string_view input, string_view originalName, string_view newName) {
+    vector<string> dotted = absl::StrSplit(input, ".");
+    // TODO: check for exactly one match
+    dotted[dotted.size() - 1] = absl::StrReplaceAll(dotted[dotted.size() - 1], {{originalName, newName}});
```

Comment:
Your question made me realize this code is just wrong -- in case of a send the loc contains the full method args, so this "split by dots" thing is too simple. It needs to properly parse the function name out of an arbitrary method call. I'll update with a fix.